### ISSUE TITLE
$('select').selectpicker('mobile') not enabling native select dropdown. 

### ISSUE THREAD

**Issue by felipearosemena** [silviomoreto/bootstrap-select#1297]
I'm trying to get the native select dropdown to work on mobile devices, but I seem to just get the standard plugin behaviour (HTML based dropdown, not native).

I'm calling the plugin like this: `$('select').selectpicker('mobile')`.

From digging through the source I found that passing the `'mobile'` parameter causes the plugin to call the mobile method, which in turn adds a `'mobile-device'` class to the element. From there I couldn't figure out how the plugin uses that to trigger the `<select>` element to open it's native menu.

**Comment by caseyjhol** [comment]
Are you wrapping it in `$(window).on('load')`? It's not documented, but `$.fn.selectpicker.Constructor.DEFAULTS.mobile = true;` is what I'm using in my implementation.

**Comment by felipearosemena** [comment]
No, I'm wasn't wrapping it in a load event. Is that a requirement?

Also, not sure I follow what you mean that's what you're using in your implementation. I'm using NPM to load the script (and importing it using ES6-Babel, then transpiling with Browserify), not sure if that helps.

In any case, are you in fact able to see the Native dropdown UI on mobile devices? How does the plugin trigger the native UI behaviour? All I can tell so far is that a mobile-device class gets added (which I confirmed is happening on my end), but the i tried to find references to that class in the plugin to no avail. I'm just confused as to how the script is supposed to be triggering the native dropdown UI to open (as opposed to opening the dynamically generated HTML UI).

**Comment by caseyjhol** [comment]
The `mobile-device` class is applied to the select itself. This class makes the select the same width and height as the button, places the select *above* the button (via z-index), and then hides it by setting the opacity to 0. So, it looks like a button, but when you click on it, you're actually clicking on the select, thus opening the native select menu.

**Comment by felipearosemena** [comment]
Du-doy! That makes total sense now. I was using my own custom styles, totally missed adding the styles for the `mobile-device` class. Thanks a lot for you help @caseyjhol !

**Comment by yCodeTech** [comment]
I am having this same issue with the latest release (1.13.18) on Android Chrome. I've tried all suggestions in this and other related issues, but the native dropdown will not appear. The plugin's dropdown html is used instead. Any help would be appreciated.

Edit:

Actually nevermind, I found that I had previously added custom css that overridden the z-index, thus stopped it from ever being clicked and triggering the native dropdown.
In addition, the documentation website could do with an overhaul and adding all these little things to the docs that are currently undocumented as you've previously stated. That would be very much helpful into determining future issues ourselves.
